providers/rke2-windows.go

```diff
@@ -0,0 +1,33 @@
+package providers
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const RKE2_WINDOWS = "rke2.windows"
+
+
+func IsRKE2Windows(ctx context.Context, k8sClient kubernetes.Interface) (bool, error) {
+	// Look for nodes that have a Windows specific label
```

Comment:
This comment should be updated to explain the rke2 check then duplicated down below

Reply:
updated w/ comments